conda_forge_tick/migrators_types.pyi

```diff
@@ -0,0 +1,134 @@
+import typing
+from typing import Any, Dict, List, Set, Tuple, Union, Optional
+
+from mypy_extensions import TypedDict
+
+PackageName = typing.NewType("PackageName", str)
```

Comment:
For my own education, what is the internal logic behind defining a special type in the typing system for a package name (which is covered by `str` already)?

Reply:
@mariusvniekerk might have more info on this than me, but it might be so we can delineate between regular strings and package names. Package names are special, since they must be in the set of package names we actually have, rather than any particular value.

Reply:
basically it allows us to have special strings of various kinds.  this is pretty useful since lots of thing are strings, but they are not the same kind of thing.  So some common ones that make sense to have special types.

PackageName
VersionNumber
Hash
Url

Reply:
I guess the typing system does static checks of this?